tests/container/BUILD

```diff
@@ -576,6 +583,16 @@ container_push(
     tags = ["manual"],
 )
 
+container_push(
```

Comment:
This container push won't actually run in the tests because you need to `bazel run` a container_push target to actually push the image. I suggest creating a `container_test` target instead and look for an expected file. This was we ensure the layer can be loaded correctly.

Reply:
The reason I used `container_push` is that other rules (eg `container_test`) use the uncompressed tarball. I verified this by changing the gzip command in `zip.bzl` and running `bazel test -s tests/container:structure_test`, which shows that gzip is not invoked when running `container_test`. Even more surprisingly, running `bazel build tests/container/compression_gzip_fast-layer.tar.gz` fails with `target not declared`, but `bazel build tests/container:new_push_compression_gzip_fast` results in creation of `compression_gzip_fast-layer.tar.gz`.

I've realized that this test is quite useless without checking that the flags affect the image size. I'm thinking of adding a testing that asserts `$(wc -c compression_gzip_normal-layer.tar.gz) -lt $(wc -c compression_gzip_fast-layer.tar.gz)`: where should I add this? Will I need to create new yaml/sh files under testing/e2e/?

Reply:
Another option is to build two versions of the same image using different compression levels and load the image in the python test [here](https://github.com/bazelbuild/rules_docker/blob/master/container/image_test.py). In this test, you could implement the equivalent check to `$(wc -c compression_gzip_normal-layer.tar.gz) -lt $(wc -c compression_gzip_fast-layer.tar.gz)`.

Reply:
Thanks for the tip! PTAL